src/overlays/actors/ovl_En_Death/z_en_death.c

```diff
@@ -50,230 +69,1683 @@ ActorInit En_Death_InitVars = {
+    EnDeath* this = THIS;
+    ArrowLight* lightArrow;
+    s32 pad;
+
+    if (this->lightArrowDamageTimer > 0) {
+        this->lightArrowDamageTimer--;
+
+        lightArrow = (ArrowLight*)SubS_FindActor(play, NULL, ACTORCAT_ITEMACTION, ACTOR_ARROW_LIGHT);
+        if (lightArrow != NULL) {
+            lightArrow->screenFillIntensity = -100.0f;
+        }
+
+        if (this->lightArrowDamageTimer == 0) {
+            play->envCtx.lightSettingOverride = 26;
+        }
+    }
+
+    EnDeath_UpdateDamage(this, play);
+    EnDeath_UpdateCore(this, play);
+    if (this->actionFunc != EnDeath_SpinAttack) {
+        Math_ScaledStepToS(&this->cloakUpperRotationModifier, 0, 0x800);
+        Math_ScaledStepToS(&this->cloakLowerRotationModifier, 0, 0x800);
+    }
+
+    this->actionFunc(this, play);
+
+    if (this->actionFunc != EnDeath_Damaged) {
+        this->actor.world.rot.y = this->actor.shape.rot.y;
+    }
+    Actor_MoveWithGravity(&this->actor);
+
+    Actor_UpdateBgCheckInfo(play, &this->actor, 40.0f, (this->actionFunc == EnDeath_SpinAttack) ? 50.0f : 100.0f, 40.0f,
+                            4 | 2 | 1);
+
+    this->bodyCollider.dim.pos.x = this->actor.world.pos.x + Math_SinS(this->actor.shape.rot.y) * 3.0f;
+    this->bodyCollider.dim.pos.z = this->actor.world.pos.z + Math_CosS(this->actor.shape.rot.y) * 3.0f;
+    this->bodyCollider.dim.pos.y = this->actor.world.pos.y;
+
+    if (this->actor.params < 5) {
+        this->coreCollider.dim.worldSphere.radius = this->coreCollider.dim.modelSphere.radius + 5;
+    } else {
+        this->coreCollider.dim.worldSphere.radius = this->coreCollider.dim.modelSphere.radius;
+    }
+
+    if (this->actionFunc != EnDeath_Dead) {
+        CollisionCheck_SetOC(play, &play->colChkCtx, &this->bodyCollider.base);
+    }
+    if (this->coreCollider.base.acFlags & AC_ON) {
+        CollisionCheck_SetAC(play, &play->colChkCtx, &this->coreCollider.base);
+    }
+    if (this->weaponCollider.base.atFlags & AT_ON) {
+        CollisionCheck_SetAT(play, &play->colChkCtx, &this->weaponCollider.base);
+    }
+    if (this->actionFunc == EnDeath_BlockProjectile || this->actionFunc == EnDeath_SpinAttack) {
+        CollisionCheck_SetAT(play, &play->colChkCtx, &this->weaponSpinningCollider.base);
+        CollisionCheck_SetAC(play, &play->colChkCtx, &this->weaponSpinningCollider.base);
+    }
+
+    if (this->dmgEffectAlpha > 0.0f) {
+        if (this->dmgEffect != ACTOR_DRAW_DMGEFF_FROZEN_NO_SFX) {
+            Math_StepToF(&this->dmgEffectAlpha, 0.0f, 0.05f);
+            this->dmgEffectScale = (this->dmgEffectAlpha + 1.0f) * 0.4f;
+
+            this->dmgEffectScale = CLAMP_MAX(this->dmgEffectScale, 0.8f);
+        } else if (Math_StepToF(&this->dmgEffectSteamScale, 0.8f, 0.02f) == 0) {
+            Actor_PlaySfx_Flagged(&this->actor, NA_SE_EV_ICE_FREEZE - SFX_FLAG);
+        }
+    }
+}
+
+void EnDeath_DrawScytheSpinning(EnDeath* this, PlayState* play) {
+    s32 i;
+    Gfx* dl;
```

Comment:
```suggestion
    Gfx* gfx;
```
Kind of personal preference. Same for the other draw functions.